packages/core-react/src/components/Button/index.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+
+export default function Button({
+  Tag = 'button',
+  children,
+  type,
+  isCompact,
+  isDanger,
+  disabled,
+  ...props
+}) {
+  return (
+    <Tag
+      {...props}
+      className={clsx('ray-button', `ray-button--${type}`, {
+        'ray-button--compact': isCompact,
+        'ray-button--danger': isDanger
+      })}
+      disabled={disabled}
+    >
+      {children}
+    </Tag>
+  );
+}
+
+Button.propTypes = {
+  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
```

Comment:
We can talk some more about Typescript vs Proptypes, but for the time being I think this is fine. I've incorporated Typescript because my team and the projects consuming our components also use TS (however, you can totally use it even if your app doesn't use TS). 

I've found using TS helpful because it's actually helped with debugging purposes while creating functional components. 

You might like this article: https://medium.com/@jtomaszewski/why-typescript-is-the-best-way-to-write-front-end-in-2019-feb855f9b164

I found some articles that go over the pros and cons of using TypeScript as I would't do great justice in explaining

Reply:
Don't think typescript is really necessary here, I'm not a fan of the bloat. Thank you for the suggestion though.